### PR TITLE
fix setcolor

### PR DESCRIPTION
**Pull request by 2youyou2**
fix setcolor

### CODE REVIEW

File: extensions/CocoStudio/Armature/utils/CCDataReaderHelper.cpp [hunk 1]
```diff
@@ -854,26 +854,26 @@ MovementBoneData *DataReaderHelper::decodeMovementBone(tinyxml2::XMLElement *mov
...
-	//! Change rotation range from (-180 -- 180) to (-infinity -- infinity)
-	CCFrameData **frames = (CCFrameData **)movBoneData->frameList.data->arr;
-	for (int i = movBoneData->frameList.count() - 1; i >= 0; i--)
-	{
-		if (i > 0)
-		{
-			float difSkewX = frames[i]->skewX -  frames[i - 1]->skewX;
-			float difSkewY = frames[i]->skewY -  frames[i - 1]->skewY;
+    //! Change rotation range from (-180 -- 180) to (-infinity -- infinity)
+    CCFrameData **frames = (CCFrameData **)movBoneData->frameList.data->arr;
```
Comment (dumganhar) [inline extensions/CocoStudio/Armature/utils/CCDataReaderHelper.cpp]: Should use `FrameData` without `CC` prefix.

**Comment by dumganhar** [comment]
Please resolve the conflict.

File: cocos/editor-support/cocostudio/CCArmature.cpp [hunk 2]
```diff
@@ -464,7 +462,7 @@ void Armature::update(float dt)
 
 void Armature::draw()
 {
-    if (_parentBone == NULL)
+    if (_parentBone == NULL && _batchNode == NULL)
```
Comment (minggo) [inline cocos/editor-support/cocostudio/CCArmature.cpp]: NULL -> nullptr.

File: cocos/editor-support/cocostudio/CCArmature.h [hunk 3]
```diff
@@ -173,6 +174,10 @@ class  Armature : public cocos2d::NodeRGBA, public cocos2d::BlendProtocol
     
     virtual bool getArmatureTransformDirty();
 
+    virtual cocos2d::TextureAtlas *getTexureAtlasWithTexture(cocos2d::Texture2D *texture);
+
+    virtual void setColliderFilter(ColliderFilter *filter);
```
Comment (minggo) [inline cocos/editor-support/cocostudio/CCArmature.h:179]: Please add `override` if possible.

File: cocos/editor-support/cocostudio/CCArmature.h [hunk 4]
```diff
@@ -173,6 +174,10 @@ class  Armature : public cocos2d::NodeRGBA, public cocos2d::BlendProtocol
     
     virtual bool getArmatureTransformDirty();
 
+    virtual cocos2d::TextureAtlas *getTexureAtlasWithTexture(cocos2d::Texture2D *texture);
```
Comment (minggo) [inline cocos/editor-support/cocostudio/CCArmature.h]: Const?

File: cocos/editor-support/cocostudio/CCArmature.h [hunk 5]
```diff
@@ -197,7 +202,7 @@ class  Armature : public cocos2d::NodeRGBA, public cocos2d::BlendProtocol
 
     CC_SYNTHESIZE(cocos2d::TextureAtlas *, _atlas, TextureAtlas);
 
-    CC_SYNTHESIZE(Bone *, _parentBone, ParentBone);
+    CC_PROPERTY(Bone *, _parentBone, ParentBone);
```
Comment (minggo) [inline cocos/editor-support/cocostudio/CCArmature.h]: Please don't use CC_PROPERTY or CC_SYNTHESIZE.

File: cocos/editor-support/cocostudio/CCArmature.cpp (hunk 2 above)
Comment (2youyou2) [inline cocos/editor-support/cocostudio/CCArmature.cpp]: ok, i see

File: cocos/editor-support/cocostudio/CCArmatureDataManager.cpp [hunk 6]
```diff
@@ -97,10 +111,45 @@ bool ArmatureDataManager::init()
...
+
+        for (std::vector<std::string>::iterator i = data->textures.begin(); i != data->textures.end(); i++)
+        {
+            removeTextureData(i->c_str());
+        }
+
+        for (std::vector<std::string>::iterator i = data->plistFiles.begin(); i != data->plistFiles.end(); i++)
+        {
+            SpriteFrameCache::getInstance()->removeSpriteFramesFromFile(i->c_str());
+        }
```
Comment (minggo) [inline cocos/editor-support/cocostudio/CCArmatureDataManager.cpp:136]: May be you can use for( : ) instead.

File: cocos/editor-support/cocostudio/CCArmature.h (hunk 5 above)
Comment (2youyou2) [inline cocos/editor-support/cocostudio/CCArmature.h]: Do i need to remove all CC_SYNTHESIZE macro?

File: cocos/editor-support/cocostudio/CCBatchNode.cpp [hunk 7]
```diff
@@ -61,9 +82,58 @@ void BatchNode::addChild(Node *child, int zOrder, int tag)
...
+
+        Dictionary *dict = armature->getBoneDic();
+        DictElement *element = NULL;
+        CCDICT_FOREACH(dict, element)
+        {
+            Bone *bone = static_cast<Bone*>(element->getObject());
+
+            Array *displayList = bone->getDisplayManager()->getDecorativeDisplayList();
+            Object *object = NULL;
+            CCARRAY_FOREACH(displayList, object)
```
Comment (minggo) [inline cocos/editor-support/cocostudio/CCBatchNode.cpp]: Please use range-based loop.

File: cocos/editor-support/cocostudio/CCBone.h [hunk 8]
```diff
@@ -140,20 +145,22 @@ class Bone : public cocos2d::NodeRGBA
     /*
      * Whether or not the bone's transform property changed. if true, the bone will update the transform.
      */
-    virtual void setTransformDirty(bool dirty);
-
-    virtual bool isTransformDirty();
+    virtual inline void setTransformDirty(bool dirty) { _boneTransformDirty = dirty; }
+    virtual inline bool isTransformDirty() { return _boneTransformDirty; }
```
Comment (minggo) [inline cocos/editor-support/cocostudio/CCBone.h]: It is meaningless to make a virtual function inline.

File: cocos/editor-support/cocostudio/CCArmature.h (hunk 5 above)
Comment (minggo) [inline cocos/editor-support/cocostudio/CCArmature.h]: Yes. Please.
Now engine codes don't use it any more.
Keep it just to keep compatibility.

File: cocos/editor-support/cocostudio/CCArmature.h (hunk 4 above)
Comment (2youyou2) [inline cocos/editor-support/cocostudio/CCArmature.h]: This is used by CCSkin,
const *TextureAtlas can not pass to  CCSprite::setTextureAtlas(TextureAtlas *pobTextureAtlas)

**Comment by dumganhar** [comment]
Please pull the latest codes on develop branch, and fix the compilation error. Thanks.

**Pull request merged**